app/src/main/java/org/simple/clinic/contactpatient/RecordPatientAgreedToVisit.kt

```diff
@@ -0,0 +1,16 @@
+package org.simple.clinic.contactpatient
+
+import org.simple.clinic.overdue.Appointment
+import org.simple.clinic.overdue.AppointmentRepository
+import org.simple.clinic.util.UserClock
+import javax.inject.Inject
+
+class RecordPatientAgreedToVisit @Inject constructor(
+    private val appointmentRepository: AppointmentRepository,
+    private val userClock: UserClock
+) {
+
+  fun execute(appointment: Appointment) {
+    appointmentRepository.markAsAgreedToVisit(appointment.uuid, userClock)
```

Comment:
This removes a few dependencies from the effect handler. Is our goal here to increase readability with extraction or something else?

Reply:
Sort of. The goal is to extract related logic into an atomic flow so that it can be reused if needed. Here, this was done because the updating of the appointment and creating the call result should be a single operation.